lib/flowViewport/sceneIndex/fvpPruneTexturesSceneIndex.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "flowViewport/sceneIndex/fvpPruneTexturesSceneIndex.h"
+#include "pxr/base/tf/staticTokens.h"
+#include "pxr/imaging/hd/sceneIndexPrimView.h"
+#include "pxr/imaging/hd/materialSchema.h"
+
+#include <iostream>
+namespace FVP_NS_DEF {
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+    (UsdPreviewSurface)
+    (ND_standard_surface_surfaceshader)
+);
+
+namespace {
+
+void
+_PruneTexturesFromMatNetwork(
+    HdMaterialNetworkInterface *networkInterface)
+{
+    if (!networkInterface) {
+        return;
+    }
+    const TfTokenVector nodeNames = networkInterface->GetNodeNames();
+    for (TfToken const &nodeName : nodeNames) {
+        const TfToken nodeType = networkInterface->GetNodeType(nodeName);
+        if (nodeType == _tokens->ND_standard_surface_surfaceshader ||
+            nodeType == _tokens->UsdPreviewSurface) {                
+            // Look for incoming connection(textures) to surface shader params
+            TfTokenVector inputConnections = networkInterface->GetNodeInputConnectionNames(nodeName);
+            for (TfToken const &connection : inputConnections) {
+                // Trivially  input connections.
+                networkInterface->DeleteNodeInputConnection(nodeName, connection);
```

Comment:
How do you know it's a texture node that you delete ? Could it be another connection type like another node ? I think we don't support network graph yet but if that come in the future, I would double check that the input connection is a texture before deleting it.

Reply:
Maya's VP2 texture toggle behavior is such that it disables *any* incoming connection to the surface shader. Whether the incoming connection is a file texture node or procedural node(ramp/checker etc) or any other. So the toggling behaviour should disable any input connections based on Maya's native behavior. What do you think?